src/lib.rs

```diff
@@ -1,203 +1,260 @@
+//! # clap
```

Comment:
I've actually (as of a commit I'm making now) making some changes by removing some sections that aren't relevant to the docs. Other than that it will still be a straight copy paste of the readme. We used to use a Makefile to update the docs, but that has since broken and I haven't bothered investigating because updating these docs is somewhat rare.

Also, the examples in the readme cause rustdoc to fail, so just including the `README.md` verbatim through `rustdoc` flags doesn't work either :\

Perhaps we can find a better way in the future, but for now this should be ok for a merge.